### PR TITLE
Network integrations tests with txpool task

### LINKED ISSUE #2014: Improve network with txpool test coverage
### Describe the feature

https://github.com/paradigmxyz/reth/pull/1901 surfaced several bugs in the network <-> pool integration that went unnoticed until #2004

to catch these, test coverage with installed txpool service needs to be improved

see existing tests in https://github.com/paradigmxyz/reth/tree/main/crates/net/network/tests/it

## TODO

* add new `network` integration tests with txpool task
* ensure transactions can be exchanged properly

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by i-m-aditya**
Fixes #2014

### CODE REVIEW

**Comment by i-m-aditya** [comment]
@mattsse Will add more tests, just want to make sure its in the right direction.

File: crates/net/network/src/transactions.rs [hunk 1]
```diff
@@ -81,7 +81,7 @@ impl TransactionsHandle {
 #[must_use = "Manager does nothing unless polled."]
 pub struct TransactionsManager<Pool> {
     /// Access to the transaction pool.
-    pool: Pool,
+    pub pool: Pool,
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:84]: all of these should remain private

File: crates/net/network/tests/it/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    let input = hex::decode("02f871018302a90f808504890aef60826b6c94ddf4c5025d1a5742cf12f74eec246d4432c295e487e09c3bbcc12b2b80c080a0f21a4eacd0bf8fea9c5105c543be5a1d8c796516875710fafafdf16d16d8ee23a001280915021bb446d1973501a67f93d2b38894a514b976e7b46dc2fe54598d76").unwrap();
+    let signed_tx = TransactionSigned::decode(&mut &input[..]).unwrap();
+    transactions.on_network_tx_event(NetworkTransactionEvent::IncomingTransactions {
+        peer_id: handle1.peer_id().clone(),
+        msg: Transactions(vec![signed_tx.clone()]),
+    });
+    assert_eq!(
+        *handle1.peer_id(),
+        transactions.transactions_by_peers.get(&signed_tx.hash()).unwrap()[0]
+    );
```
Comment (mattsse) [inline crates/net/network/tests/it/transaction.rs:88]: there's a trait function that returns the tx by hash.

**Review by mattsse — CHANGES_REQUESTED** [review]
nice tests.

all the fields and private functions should remain private.
the `tests/integrationtests` should only ever use public interfaces, perhaps some things need to be exposed but it test should not invoke private functions directly.
if we need to test these then we can add the test inside the module see, txpool.rs

**Review by mattsse — APPROVED** [review]
nice, tests looking good,

made a smol change in the network config so port can be configured and discovery is always disabled.

**Pull request merged**